src/vs/workbench/contrib/chat/browser/chatInputPart.ts

```diff
@@ -185,13 +186,11 @@ export class ChatInputPart extends Disposable implements IHistoryNavigationWidge
 		// prompt files may have nested child references to other prompt
 		// files that are resolved asynchronously, hence we need to wait
 		// for the entire prompt instruction tree to be processed
-		const instructionsStarted = performance.now();
-
-		// wait for all prompt files resolve precesses to settle
-		await this.promptInstructionsAttachmentsPart.allSettled();
-
-		// allow-any-unicode-next-line
-		this.logService.trace(`[⏱] instructions tree resolved in ${performance.now() - instructionsStarted}ms`);
+		await logExecutionTime('instructions tree resolve',
```

Comment:
I suggest using the `ITimerService` for this which lets you record and log performance marks.

Reply:
Humm the `ITimerService` seem to be used for another purpose - to measure the `startup` times so I cannot find any example that will fit the case above. This function can be called multiple times during the session, and we are not interested in relative time when compared with the startup (only time to execute the function itself) 🤔

Reply:
Oh, sorry, it looks like ITimerService is generally downstream, components actually use `mark` from `src/vs/base/common/performance` and then the ITimerService just collects those marks

Reply:
Right, all the examples I've found are related to recording components' initialization times relating to the common start time of the workbench. E.g., they are one-off and always measure time from the overall start 🤔 Or am I missing something?